src/query/expression/src/kernels/take.rs

```diff
@@ -144,6 +173,244 @@ impl Column {
+                                    col_ptr.add(index as usize),
+                                    builder_ptr.add(offset),
+                                    1,
+                                );
+                            }
+                            offset += 1;
+                            continue;
+                        }
+                        // Using the doubling method to copy memory.
+                        let base_offset = offset;
+                        unsafe {
+                            std::ptr::copy_nonoverlapping(
+                                col_ptr.add(index as usize),
+                                builder_ptr.add(base_offset),
+                                1,
+                            );
+                        }
+                        remain = cnt as usize;
+                        // Since cnt > 0, then 31 - cnt.leading_zeros() >= 0.
+                        let max_segment = 1 << (31 - cnt.leading_zeros());
+                        let mut cur_segment = 1;
+                        while cur_segment < max_segment {
+                            unsafe {
+                                std::ptr::copy_nonoverlapping(
+                                    builder_ptr.add(base_offset),
+                                    builder_ptr.add(base_offset + cur_segment),
+                                    cur_segment,
+                                );
+                            }
+                            cur_segment <<= 1;
+                        }
+                        remain -= max_segment;
+                        offset += max_segment;
+                        power = 0;
+                        while remain > 0 {
+                            if remain & 1 == 1 {
+                                let cur_segment = 1 << power;
+                                unsafe {
+                                    std::ptr::copy_nonoverlapping(
+                                        builder_ptr.add(base_offset),
+                                        builder_ptr.add(offset),
+                                        cur_segment,
+                                    );
+                                }
+                                offset += cur_segment;
+                            }
+                            power += 1;
+                            remain >>= 1;
+                        }
+                    }
+                    unsafe {
+                        builder.set_len(offset);
+                    }
+                    Column::Decimal(DecimalColumn::Decimal256(builder.into(), *size))
+                }
+            }),
+            Column::Boolean(bm) => {
+                Self::probe_take_arg_types::<BooleanType>(bm, indices, indices_len, probe_num)
+            }
+            Column::String(column) => {
+                Self::probe_take_arg_types::<StringType>(column, indices, indices_len, probe_num)
+            }
+            Column::Timestamp(column) => {
```

Comment:
ditto

Reply:
numeric path will call probe_take_arg_types and will have the same behavior as decimal.